exp/api/v1beta1/azuremachinepool_default.go

```diff
@@ -159,3 +160,13 @@ func (amp *AzureMachinePool) SetNetworkInterfacesDefaults() {
 		}
 	}
 }
+
+// SetOSDiskDefaults sets the defaults for the OSDisk.
+func (amp *AzureMachinePool) SetOSDiskDefaults() {
+	if amp.Spec.Template.OSDisk.OSType == "" {
```

Comment:
this is not done for AzureMachine. There wasn't a OSDisk validation before (not a lot of things are validated for AzureMachinePools as opposed to AzureMachines. Wondering if there should be a refactoring done to improve those?

Reply:
I think OSType is a required field in OSDisk so this shouldn't be necessary: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a52056dfb88c1507e8a13817ddcf502beff3a015/api/v1beta1/types.go#L599

Reply:
true, however currently the validation is not done and existing MachinePool YAMLs break if they do not specify OSType. 

I think it works because VMSS defaults to linux if not specified. 

How should we handle this?

Reply:
Ah that makes sense, I think we should leave it as is if it ends up breaking existing MachinePools.